pkg/api/rest/delete.go

```diff
@@ -89,5 +92,13 @@ func BeforeDelete(strategy RESTDeleteStrategy, ctx api.Context, obj runtime.Obje
 	now := unversioned.NewTime(unversioned.Now().Add(time.Second * time.Duration(*options.GracePeriodSeconds)))
 	objectMeta.DeletionTimestamp = &now
 	objectMeta.DeletionGracePeriodSeconds = options.GracePeriodSeconds
+	// If it'a first graceful deletion we're going to update the object first,
+	// setting DeletionTimestamp to something != nil. Object that's being deleted shouldn't
+	// take any nontrivial actions, hence it's behavior changes. Thus we need to bump object's
+	// Generation (if set). This handles generation bump during graceful deletion. For bump
+  // caused by the finalizer-driven cascading deletion go to pkg/registry/generic/registry/store.go
+	if objectMeta.Generation > 0 {
```

Comment:
If the strategy updates the generation this could potentially bump twice.  I'm not sure we'd want the strategy to bump generation, but something we should all remember in the future.